src/app/modules/form/admin-form/admin-form.service.ts

```diff
@@ -1148,3 +1170,315 @@ const isMobileFieldUpdateAllowed = (
+ * @returns err(SecretsManagerError) if an error occurs while creating credentials in secrets manager
+ */
+export const createTwilioCredentials = (
+  twilioCredentials: TwilioCredentials,
+  form: IPopulatedForm,
+): ResultAsync<
+  unknown,
+  ReturnType<typeof transformMongoError> | SecretsManagerError
+> => {
+  const twilioCredentialsData: TwilioCredentialsData =
+    new TwilioCredentialsData(twilioCredentials)
+  const formId = form._id
+
+  const msgSrvcName = generateTwilioCredSecretKeyName(formId)
+
+  const body: CreateSecretRequest = {
+    Name: msgSrvcName,
+    SecretString: twilioCredentialsData.toString(),
+    Description: `autogenerated via API on ${new Date().toISOString()} by ${
+      form.admin._id
+    }`,
+  }
+
+  const logMeta = {
+    action: 'createTwilioCredentials',
+    formId: formId,
+    msgSrvcName,
+    body,
+  }
+
+  logger.info({
+    message: `No msgSrvcName, creating Twilio credentials for form ${formId}`,
+    meta: logMeta,
+  })
+
+  return ResultAsync.fromPromise(
+    FormModel.startSession().then((session: ClientSession) =>
+      session
+        .withTransaction(() =>
+          createTwilioTransaction(form, msgSrvcName, body, session),
+        )
+        .then(() => session.endSession()),
+    ),
+    (error) => {
+      logger.error({
+        message: 'Error encountered when creating Twilio Secret',
+        meta: logMeta,
+        error,
+      })
+
+      return error as
+        | ReturnType<typeof transformMongoError>
+        | SecretsManagerError
+    },
+  )
+}
+
+/**
+ * Updates msgSrvcName of the form in the database, uses the msgSrvcName as the
+ * key to store the Twilio Credentials in AWS Secrets Manager
+ * @param form The form to add Twilio Credentials
+ * @param msgSrvcName The key under which the credentials is stored in AWS Secrets Manager
+ * @param body the request body used to create the secret in secrets manager
+ * @param session session of the transaction
+ * @returns Promise.ok(void) if the creation is successful
+ */
+// Exported to use in tests
+export const createTwilioTransaction = async (
+  form: IPopulatedForm,
+  msgSrvcName: string,
+  body: CreateSecretRequest,
+  session: ClientSession,
+): Promise<void> => {
+  const meta = {
+    action: 'createTwilioTransaction',
+    formId: form._id,
+    msgSrvcName,
+    body,
+  }
+
+  try {
+    await form.updateMsgSrvcName(msgSrvcName, session)
+  } catch (err) {
+    logger.error({
+      message:
+        'Error occured when updating msgSrvcName, rolling back transaction!',
+      meta,
+      error: err,
+    })
+    throw transformMongoError(err)
+  }
+
+  try {
+    await secretsManager.createSecret(body).promise()
+  } catch (err) {
+    const awsError = err as AWSError
+
+    logger.error({
+      message:
+        'Error occured when creating secret AWS Secrets Manager, rolling back transaction!',
+      meta,
+      error: awsError,
+    })
+    throw new SecretsManagerError(awsError.message)
+  }
+}
```

Comment:
@karrui This was what I meant when splitting the 2 operations into 2 try catch blocks for better error handling and logging of AWS and Mongo Errors. 

Conventionally, we should be using multiple catch blocks (and 1 try block), but in this case, there seems no way to check if the error is an instance of an AWSError (just a type) or MongoError.

Any idea how to improve on this?